fetch-in-react-practice/client/src/components/Todos.tsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+/* eslint-disable @typescript-eslint/no-empty-function -- Remove me */
+import { useEffect, useState } from 'react';
+import { PageTitle } from './PageTitle';
+import { TodoList } from './TodoList';
+import { TodoForm } from './TodoForm';
+
+export type UnsavedTodo = {
+  task: string;
+  isCompleted: boolean;
+};
+export type Todo = UnsavedTodo & {
+  todoId: number;
+};
+
+export function Todos() {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    setIsLoading(true);
+    async function load() {
+      try {
+        const res = await fetch('/api/todos');
+        if (!res.ok) throw new Error(res.statusText);
+        const todoList = await res.json();
+        setTodos(todoList);
+      } catch (Error) {
+        setError('new note' + Error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    load();
+    console.log(todos);
+  }, []);
+
+  /* Implement addTodo to add a new todo. Hints are at the bottom of the file. */
+  async function addTodo(newTodo: UnsavedTodo) {
+    try {
+      const res = await fetch(`/api/todos`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(newTodo),
+      });
+      if (!res.ok) throw new Error(res.statusText);
+      const newTask: Todo = await res.json();
+      const newTasks: Todo[] = [];
+      todos.map((element) => {
```

Comment:
`.map` returns an array of the new elements. You are using it here to simply loop over the elements. If you just want to loop over elements in an array you should use `.forEach`.